src/components/Quickstart/TableOfContents/index.tsx

```diff
@@ -0,0 +1,48 @@
+/** @jsxImportSource preact */
+import type { FunctionalComponent } from "preact"
+import { useRef } from "preact/hooks"
+import { useCurrentIds } from "../../../hooks/currentIds/useCurrentIds"
+import { MarkdownHeading } from "astro"
+import styles from "./tableOfContents.module.css"
+
+const TableOfContents: FunctionalComponent<{
+  headings: MarkdownHeading[]
+}> = ({ headings }) => {
```

Comment:
This is a good idea (for the rest of docs ToC as well)... And it gets me thinking about more of a global variable we could refer to as `depthLevel` since the section wrapper script also refers to H1 and H2 as being parent elements.